source/d1clx.cpp

```diff
@@ -0,0 +1,17 @@
+#include "d1clx.h"
+
+quint16 D1ClxFrame::computeWidthFromHeader(QByteArray &rawFrameData)
+{
+    QDataStream in(rawFrameData);
+    in.setByteOrder(QDataStream::LittleEndian);
+    in.skipRawData(2);
+    quint16 result;
+    in >> result;
+    in.skipRawData(6);
```

Comment:
nit: is this needed

Reply:
Yes, it is expected that the input is past the header after calling this function

Reply:
Oh, I didn't think the pointer survived past the function call since `QDataStream` was created inside of it.